app/src/androidTest/java/com/github/ybecker/epforuml/NewQuestionTest.kt

```diff
@@ -1,70 +1,116 @@
 package com.github.ybecker.epforuml
 
-import android.app.Activity
 import android.content.Intent
-import android.graphics.Color
 import androidx.test.core.app.ActivityScenario
 import androidx.test.core.app.ApplicationProvider
 import androidx.test.espresso.Espresso.onView
-import androidx.test.espresso.NoActivityResumedException
 import androidx.test.espresso.assertion.ViewAssertions.matches
 import androidx.test.espresso.matcher.ViewMatchers.*
 import androidx.test.ext.junit.runners.AndroidJUnit4
-import com.dsphotoeditor.sdk.activity.DsPhotoEditorActivity
-import com.dsphotoeditor.sdk.utils.DsPhotoEditorConstants
-import android.net.Uri
 import androidx.test.espresso.action.ViewActions.*
-import androidx.test.espresso.intent.Intents
 import com.github.ybecker.epforuml.authentication.LoginActivity
 import com.github.ybecker.epforuml.database.DatabaseManager
-import org.junit.After
-import org.junit.Assert.assertThrows
 import org.junit.Before
-import org.junit.Rule
-import org.junit.Test import org.junit.rules.ExpectedException
+import org.junit.Test
 import org.junit.runner.RunWith
 
 import androidx.test.espresso.Espresso.onData
-import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions
 import androidx.test.espresso.action.ViewActions.click
 import androidx.test.espresso.action.ViewActions.typeText
 import androidx.test.espresso.assertion.ViewAssertions
-import androidx.test.espresso.assertion.ViewAssertions.matches
-import androidx.test.espresso.matcher.ViewMatchers
-import androidx.test.espresso.matcher.ViewMatchers.*
 import com.github.ybecker.epforuml.database.DatabaseManager.db
-import com.github.ybecker.epforuml.database.MockDatabase
-import com.github.ybecker.epforuml.database.Model
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
-import junit.framework.TestCase.assertNotNull
+import junit.framework.TestCase.*
 import org.hamcrest.CoreMatchers.*
-import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers.hasSize
+
 @RunWith(AndroidJUnit4::class)
+
+
+
+
 class NewQuestionTest {
+    private lateinit var scenario : ActivityScenario<MainActivity>
 
     @Before
     fun setup() {
         Firebase.auth.signOut()
         DatabaseManager.useMockDatabase()
 
     }
+
+
     @Test
-    fun testAddAQuestion() {
+    fun testAddQuestionWithNullUser() {
+
+        Firebase.auth.signOut()
+        DatabaseManager.useMockDatabase()
+        DatabaseManager.user = null
+        val scenario = ActivityScenario.launch(LoginActivity::class.java)
+        // go to MainActivity
+
+        Thread.sleep(2000)
```

Comment:
This is not a good practice in tests. You should really consider to test it without sleeps.